scrypto-derive/src/blueprint.rs

```diff
@@ -166,33 +171,121 @@ pub fn handle_blueprint(input: TokenStream) -> Result<TokenStream> {
                     event_schema.insert(#event_type_names.to_owned(), local_type_index);
                 })*
 
+                let mut method_auth_template = method_auth_template();
+                if !method_auth_template.contains_key(&SchemaMethodKey::metadata(scrypto::api::node_modules::metadata::METADATA_GET_IDENT)) {
+                    method_auth_template.insert(SchemaMethodKey::metadata(scrypto::api::node_modules::metadata::METADATA_GET_IDENT), SchemaMethodPermission::Public);
+                    method_auth_template.insert(SchemaMethodKey::metadata(scrypto::api::node_modules::metadata::METADATA_SET_IDENT), [OWNER_ROLE].into());
+                    method_auth_template.insert(SchemaMethodKey::metadata(scrypto::api::node_modules::metadata::METADATA_REMOVE_IDENT), [OWNER_ROLE].into());
+                }
```

Comment:
As discussed offline, I think this exception may cause more harm than good. It breaks the model "if you don't enable it, you won't get it".

I thought Russ was kind of okay with this, after realising none of the example components is using metadata; metadata is more widely used with resources.

Reply:
Right, Russ wanted this back. But atleast here it's now more of a language concern rather than baked in the system.